File: 3xp1.c
#include <stdio.h>
#include <math.h>

int main(void) {
	long int num, n;
	// printf("Enter a number: \n");
	// scanf("%i", &num);
	num = 10;
	n = 0;
	while (n < num) {
		n++;
		int temp;
		long int nums[1000000], i;
		i = n+0.0;
		temp = 0.0;
		while (i!=1.0) {
			if (fmod(i,2.0)==0.0) {
				i= i/2.0;
			}
			else if (fmod(i,2.0)==1.0) {
				i = (3.0*i)+1.0;
			}
			nums[temp] = i+0.0;
			temp++;
		}
		printf("%.0ld: %i\n",n,temp);
		printf("[");
		int reached = 0;
		for (int x=0;x<temp;x++) {
			if (nums[x] == 4.0 || nums[x] == 2.0 || nums[x] == 1.0) {
				reached = 1;
			}
			if (reached) {
				printf("%.0ld, ",nums[x]);
			}
		}
		printf("\b\b]\n\n");
	}
}

File: 3xp1.py
import timeit
printable = []
num = int(input("Enter a Number: \n"))
print("\n")
def solve():
    n = 0
    global num, printable
    while n < num:
        n += 1
        nums = []
        i = n+0
        temp = 0
        while i!=1:
            if i%2 == 0:
                i//=2
            elif i%2 == 1:
                i = (3*i)+1
            temp += 1
            nums.append(i)
        printable.append(f"{n}: {len(nums)}\n{nums[-3:]}\n")
    
print(timeit.timeit(solve,number=1))
print("\n")
for item in printable:
    print(item)

